ksql-engine/src/test/java/io/confluent/ksql/analyzer/QueryAnalyzerTest.java

```diff
@@ -134,4 +134,29 @@ public void shouldProcessHavingExpression() {
         new QualifiedNameReference(QualifiedName.of("KSQL_AGG_VARIABLE_1")),
         new LongLiteral("10"))));
   }
+
+  @Test
+  public void shouldFailWithIncorrectJoinCriteria() {
+    final List<Statement> statements = ksqlParser.buildAst(
+        "select * from test1 join test2 on test1.col1 = test2.coll;",
+        metaStore);
+    final Query query = (Query) statements.get(0);
+    try {
+      final Analysis analysis = queryAnalyzer.analyize(query);
+    } catch (KsqlException ex) {
+      assertThat(ex.getMessage().trim(), equalTo("Line: 1, Col: 46 : Invalid join criteria "
+                                             + "(TEST1.COL1 = TEST2.COLL)"));
+    }
+  }
+
+  @Test
+  public void shouldPassJoinWithAnyCriteriaOrder() {
+    final List<Statement> statements = ksqlParser.buildAst(
+        "select * from test1 join test2 on test2.col2 = test1.col1;",
+        metaStore);
+    final Query query = (Query) statements.get(0);
+    final Analysis analysis = queryAnalyzer.analyize(query);
```

Comment:
Might want to check that the result has the correct structure

Reply:
Added join checks.